src/lib.rs

```diff
@@ -194,6 +236,160 @@ impl IntoRawFd for VsockListener {
 }
 
 /// A virtio stream between a local and a remote socket.
+///
+/// This is the vsock equivalent of [`std::net::UdpSocket`].
+#[derive(Debug)]
+pub struct VsockSocket {
+    socket: OwnedFd,
+}
+
+impl VsockSocket {
+    /// Bind to an address and listen for connections.
+    pub fn bind<A: SockaddrLike>(addr: &A) -> Result<Self> {
+        if addr.family() != Some(AddressFamily::Vsock) {
+            return Err(Error::other("requires a virtio socket address"));
+        }
+
+        let socket = new_socket(SockType::Datagram)?;
+
+        bind(socket.as_raw_fd(), addr)?;
+
+        Ok(Self { socket })
+    }
+
+    /// Bind to a remote host with specified cid and port.
+    pub fn bind_with_cid_port(&self, cid: u32, port: u32) -> Result<Self> {
+        Self::bind(&VsockAddr::new(cid, port))
+    }
+
+    /// Receive a message from a remote host.
+    ///
+    /// # Returns
+    ///
+    /// The number of bytes read and the address of the remote host.
+    pub fn recv_from(&self, buf: &mut [u8]) -> Result<(usize, VsockAddr)> {
+        nix::sys::socket::recvfrom(self.socket.as_raw_fd(), buf)
```

Comment:
Add this (and the other functions from `nix::sys::socket` to the use statement at the top of the file.